R/correlation.R

```diff
@@ -40,6 +40,33 @@
 #' @importFrom rlang .data !! !!! enquo syms
 #' @importFrom tidyselect eval_select
 #' @export
+#' @examples
+#'  
+#' # linear association of case and death rates on any given day
+#' epi_cor(x = jhu_csse_daily, 
+#'         var1 = case_rate_7d_av, 
+#'         var2 = death_rate_7d_av, 
+#'         cor_by = "time_value")
```

Comment:
Agreed. @ChloeYou can you update the data please?

Reply:
Thanks, these look more like actual correlation values at a glance (rather than some sort of binary variable).  But the negatives still need some explanation in a comment, as I think it's natural to expect positive correlations between cases and deaths.  It looks like the explanation is some combination of (a) across-all-state by-time correlations being lower in this time period, the onset of the second wave in the US (reported cases began to rise rapidly in many locations, but reported deaths had not yet reflected these increases or changes in geographical patterns), and (b) the drop in by-time correlations being even more precipitous across this selection of 4 locations than across all "state" locations.

<details>
  <summary>Exploration code</summary>

```
library(tidyverse)

time_values = epirange(20200301,20200901)

cases7rb = covidcast("jhu-csse","confirmed_7dav_incidence_prop","day","state",time_values,"*")%>%fetch_tbl()

deaths7rb = covidcast("jhu-csse","deaths_7dav_incidence_prop","day","state",time_values,"*")%>%fetch_tbl()

combined = dplyr::bind_rows(cases7rb, deaths7rb)

combined %>%
  filter(geo_value %in% unique(jhu_csse_daily_subset$geo_value)) %>%
  ggplot(aes(time_value, value, colour=geo_value)) %>%
  `+`(geom_line()) %>%
  `+`(facet_wrap(~ signal, scales="free_y")) %>%
  `+`(geom_vline(xintercept=as.Date(c("2020-05-26","2020-06-04","2020-06-28","2020-07-16")), linetype="dashed"))



combined_wider = combined %>%
  pivot_wider(id_cols=c(geo_value,time_value), names_from="signal", values_from="value")

list(
  all_geos = combined_wider,
  selected_geos = combined_wider %>% filter(geo_value %in% unique(jhu_csse_daily_subset$geo_value))
) %>%
  map_dfr(function(tbl) {
    tbl %>%
      group_by(time_value) %>%
      dplyr::summarize(cor=cor(confirmed_7dav_incidence_prop, deaths_7dav_incidence_prop), .groups="drop")
  }, .id="geo_set") %>%
  ggplot(aes(time_value, cor, colour=geo_set)) %>%
  `+`(geom_path()) %>%
  `+`(geom_vline(aes(xintercept=time_value, colour=geo_set),
                 linetype="dotdash",
                 data=function(tbl) {
                   tbl %>%
                     filter(is.na(cor))
                 })) %>%
  `+`(geom_hline(yintercept=c(-1,0,1), linetype="dashed")) %>%
  `+`(geom_vline(xintercept=as.Date(c("2020-05-26","2020-06-04","2020-06-28","2020-07-16")), linetype="dashed"))
```
</details>

Reply:
handled by dajmcdon/epiprocess#10